### ISSUE TITLE
Fix the disconnection of haar_emb_matrix from the computational graph.

### ISSUE THREAD

**Issue by angelolmn** [angelolmn/HitsBE#12]
**Solution:**  
Remove calls to `.detach()` and `.item()`. These calls disconnect `haar_embed_matrix` from the PyTorch computational graph, preventing gradient computation.
